File: 547-NumberOfProvinces/547-NumberOfProvinces.py
# Last updated: 08/05/2026, 09:29:18

class Solution:
    def findCircleNum(self, isConnected: List[List[int]]) -> int:
        N= len(isConnected)
        visited=[False]* N
        ans=0
        def dfs(citi):

            for i in range(N):
                if  isConnected[citi][i] ==1 and visited[i]==False:
                    visited[i]= True

                    neighbor = i
                    dfs(i)

        for i in range(N):
            if visited[i] ==  False:
                dfs(i)
                ans+=1

        return ans

File: 547-NumberOfProvinces/547-NumberOfProvinces.cpp
// Last updated: 08/05/2026, 09:29:14
class Solution {
public:
    vector<vector<int>> isConnected;
    
    void dfs(vector<bool> &visited, int i){
        visited[i]=true;
        for(int j=0; j<isConnected.size(); j++){
        if(isConnected[i][j]==1 && visited[j]==false){
        dfs(visited,j);    }
        }
        
    }
    
    
    int findCircleNum(vector<vector<int>>& isConnected) {
        int len=isConnected[0].size();
        vector<bool> visited(len, false);
        int ans=0;
        this->isConnected= isConnected;
        for(int i=0; i<len;i++){
            if(visited[i]==false){
                dfs(visited,i);
                ans+=1;
            }
        }
        return ans;

    }
};